demo/src/test/scala/dev/atedeg/ecscalademo/MathTest.scala

```diff
@@ -0,0 +1,79 @@
+package dev.atedeg.ecscalademo
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class MathTest extends AnyWordSpec with Matchers {
+
+  "A point" when {
+    "adding a vector" should {
+      "return a point with summed components" in {
+        Point(1, 1) + Vector(2, 3) shouldBe Point(3, 4)
+      }
+    }
+    "subtracting a vector" should {
+      "return a point with subtracted components" in {
+        Point(1, 1) - Vector(2, 3) shouldBe Point(-1, -2)
+      }
+    }
+    "subtracting a point" should {
+      "return a vector with subtracted components" in {
+        Point(2, 3) - Point(1, 1) shouldBe Vector(1, 2)
+      }
+    }
+  }
+
+  "A vector" when {
+    "adding another vector" should {
+      "return the correct result" in {
+        Vector(1, 1) + Vector(2, 2) shouldBe Vector(3, 3)
+      }
+      "subtract it" in {
+        Vector(1, 1) - Vector(2, 2) shouldBe Vector(-1, -1)
+      }
+      "compute the dot product" in {
+        Vector(1, 2) dot Vector(3, 4) shouldBe 11
+      }
```

Comment:
This should be changed, you are not `"adding another vector"`; they should be moved to their own `when`:
```suggestion
    "subtracting another vector" should {
      "return the correct result" in {
        Vector(1, 1) - Vector(2, 2) shouldBe Vector(-1, -1)
      }
    }
    "computing the dot product" should {
      "return the correct result" in {
        Vector(1, 2) dot Vector(3, 4) shouldBe 11
      }
    }
```